src/braginskii_heat_exchange.cxx

```diff
@@ -0,0 +1,132 @@
+#include <iterator>
+
+#include <bout/constants.hxx>
+#include <bout/field.hxx>
+#include <bout/output_bout_types.hxx>
+
+#include "../include/braginskii_heat_exchange.hxx"
+#include "../include/hermes_utils.hxx"
+
+BraginskiiHeatExchange::BraginskiiHeatExchange(std::string name, Options& alloptions,
+                                               Solver*) {
+  AUTO_TRACE();
+  diagnose = alloptions[name]["diagnose"]
+                 .doc("Output additional diagnostics?")
+                 .withDefault<bool>(false);
+}
+
+void BraginskiiHeatExchange::transform(Options& state) {
+  AUTO_TRACE();
+
+  Options& allspecies = state["species"];
+
+  // Iterate through all species
+  // To avoid double counting, this needs to iterate over pairs
+  // i.e. the diagonal and above
+  //
+  // Iterators kv1 and kv2 over the species map
+  //
+  //               kv2 ->
+  //             species1  species2  species3
+  // kv1   species1     X         X         X
+  //  ||   species2               X         X
+  //  \/   species3                         X
+  //
+  const std::map<std::string, Options>& children = allspecies.getChildren();
+  for (auto kv1 = std::begin(children); kv1 != std::end(children); ++kv1) {
+    Options& species1 = allspecies[kv1->first];
+    // If collisions were not calculated for this species, skip it.
+    if (not species1.isSection("collision_frequencies"))
+      continue;
+
+    const Field3D density1 = GET_NOBOUNDARY(Field3D, species1["density"]),
+                  temperature1 = species1.isSet("temperature")
+                                     ? GET_NOBOUNDARY(Field3D, species1["temperature"])
+                                     : 0.0;
+    ;
+
+    const BoutReal A1 = GET_VALUE(BoutReal, species1["AA"]),
+                   Z1 = species1.isSet("charge") ? GET_VALUE(BoutReal, species1["charge"])
+                                                 : 0;
```

Comment:
Unused?

Reply:
I guess so. This would have predated my refactoring I think.